runtime/src/iree/tooling/comparison.cc

```diff
@@ -0,0 +1,282 @@
+    case IREE_VM_VALUE_TYPE_I8:
+      if (expected_variant.i8 != actual_variant.i8) {
+        IREE_CHECK_OK(iree_string_builder_append_format(
+            builder,
+            "[FAILED] result[%d]: i8 values differ\n  expected: %" PRIi8
+            "\n  actual: %" PRIi8 "\n",
+            result_index, expected_variant.i8, actual_variant.i8));
+        return false;
+      }
+      return true;
+    case IREE_VM_VALUE_TYPE_I16:
+      if (expected_variant.i16 != actual_variant.i16) {
+        IREE_CHECK_OK(iree_string_builder_append_format(
+            builder,
+            "[FAILED] result[%d]: i16 values differ\n  expected: %" PRIi16
+            "\n  actual: %" PRIi16 "\n",
+            result_index, expected_variant.i16, actual_variant.i16));
+        return false;
+      }
+      return true;
+    case IREE_VM_VALUE_TYPE_I32:
+      if (expected_variant.i32 != actual_variant.i32) {
+        IREE_CHECK_OK(iree_string_builder_append_format(
+            builder,
+            "[FAILED] result[%d]: i32 values differ\n  expected: %" PRIi32
+            "\n  actual: %" PRIi32 "\n",
+            result_index, expected_variant.i32, actual_variant.i32));
+        return false;
+      }
+      return true;
+    case IREE_VM_VALUE_TYPE_I64:
+      if (expected_variant.i64 != actual_variant.i64) {
+        IREE_CHECK_OK(iree_string_builder_append_format(
+            builder,
+            "[FAILED] result[%d]: i64 values differ\n  expected: %" PRIi64
+            "\n  actual: %" PRIi64 "\n",
+            result_index, expected_variant.i64, actual_variant.i64));
+        return false;
+      }
+      return true;
+    case IREE_VM_VALUE_TYPE_F32:
+      // TODO(benvanik): use tolerance flag.
+      if (expected_variant.f32 != actual_variant.f32) {
+        IREE_CHECK_OK(iree_string_builder_append_format(
+            builder,
+            "[FAILED] result[%d]: f32 values differ\n  expected: %G\n  actual: "
+            "%G\n",
+            result_index, expected_variant.f32, actual_variant.f32));
+        return false;
+      }
+      return true;
+    case IREE_VM_VALUE_TYPE_F64:
+      // TODO(benvanik): use tolerance flag.
+      if (expected_variant.f64 != actual_variant.f64) {
+        IREE_CHECK_OK(iree_string_builder_append_format(
+            builder,
+            "[FAILED] result[%d]: f64 values differ\n  expected: %G\n  actual: "
+            "%G\n",
+            result_index, expected_variant.f64, actual_variant.f64));
+        return false;
+      }
+      return true;
+    default:
+      IREE_CHECK_OK(iree_string_builder_append_format(
+          builder, "[FAILED] result[%d]: unknown value type, cannot match\n",
+          result_index));
+      return false;
+  }
+}
+
+static bool iree_tooling_compare_buffer_views(
+    int result_index, iree_hal_buffer_view_t* expected_view,
+    iree_hal_buffer_view_t* actual_view, iree_allocator_t host_allocator,
+    iree_host_size_t max_element_count, iree_string_builder_t* builder) {
```

Comment:
Is it possible to use the `max_element_count` to set the compare number? For example, DeepLabV3 has 227x227x21 output, and we probably don't want to provide an expected output to check all of them.

Reply:
so you want to be able to provide subranges of the input to compare? seems useful, but that's a bit trickier as we'd need a way to specify partial inputs (say an input is 227x227x21 but that only N elements are provided and valid) - for now it'd be best to save the expected values to npy files and compare against those and we can make this more complicated in the future

Reply:
Sure, for the large files we could do so as a first step. for mobilenet or efficientnet we can still list all outputs in text (1001 elements)

It would be nicer to store in text format instead of bin/npy for reviewing purposes, but I understand the complexity. :)

For the text format, will the tool allow line breaks? Or do we need to store the `--expected-output=` in one line?

Reply:
Just to make sure: the syntax to use npy as expected output is `--expected_output=@<npy path>`, correct?